benchmark/src/bin/ruben.rs

```diff
@@ -1,33 +1,58 @@
 // Copyright (c) The Libra Core Contributors
 // SPDX-License-Identifier: Apache-2.0
 
-use admission_control_proto::proto::admission_control_grpc::AdmissionControlClient;
+use admission_control_proto::proto::{
+    admission_control::SubmitTransactionRequest, admission_control_grpc::AdmissionControlClient,
+};
 use benchmark::{
     ruben_opt::{Executable, Opt},
+    txn_generator::{
+        gen_accounts, gen_pairwise_transfer_txn_requests, gen_ring_transfer_txn_requests,
+    },
     Benchmarker,
 };
 use client::AccountData;
 use debug_interface::NodeDebugClient;
 use grpcio::{ChannelBuilder, EnvBuilder};
+use libra_wallet::wallet_library::WalletLibrary;
 use logger::{self, prelude::*};
 use metrics::metric_server::start_server;
 use std::sync::Arc;
 
+/// Helper method to generate repeated TXNs from a function object, which is soon to be
+/// replaced by generic struct that implements TxnGenerator trait.
+#[allow(bare_trait_objects)]
```

Comment:
what is this lint rule for in this case?

Reply:
This will have to be cleaned after the <soon to be replaced by> work.

Reply:
> what is this lint rule for in this case?
From "The rustc book"
> This lint suggests using dyn Trait for trait objects.
To my understanding, the linter ask me to create a trait object for this function object, which is exactly what I will do in the next PR.